### ISSUE TITLE
[LINT_BUG]: box_usage_linter false positive on local module $function access when using lint_dir()

### ISSUE THREAD

**Issue by dereckmezquita** [Appsilon/box.linters#188]
### box.linters version

0.10.7

### Sample source code to lint

Two files in a subdirectory `src/`:

**src/helper.R**
```r
#' @export
add <- function(a, b) {
  return(a + b)
}
```

**src/main.R**
```r
box::use(./helper)

test_fn <- function() {
  return(helper$add(1, 2))
}
```

### Lint command used

From the **parent directory** of `src/`:

```
Rscript -e 'lintr::lint_dir("src", linters = list(box.linters::box_usage_linter()))'
```

Output:
```
main.R:4:10: warning: [box.linters::box_usage_linter] <package/module>$function does not exist.
  return(helper$add(1, 2))
         ^~~~~~~~~~
```

From **inside** `src/`:

```
Rscript -e 'lintr::lint_dir(linters = list(box.linters::box_usage_linter()))'
```

Output:
```
No lints found.
```

### Lint result

main.R:4:10: warning: [box.linters::box_usage_linter] <package/module>$function does not exist.
  return(helper$add(1, 2))
         ^~~~~~~~~~

### Expected result

No lint. `helper$add()` is a valid call the function `add` exists and is exprted by the local module `helper.R` in the same directory.

The linter resolves local module `$function` access correctly when the working directory matches the file's directory, but produces a false positive when linting from a parent directory. Since `lint_dir()` always lints from a parent, this makes `box_usage_linter` unusable for any project that uses local module `$` access.

This may share a root cause with #137 (working directory resolution for relative imports).
